lib/commands/general.js

```diff
@@ -297,17 +297,24 @@ helpers.suspendChromedriverProxy = function suspendChromedriverProxy () {
 };
 
 commands.startActivity = async function startActivity (appPackage, appActivity,
-  appWaitPackage, appWaitActivity) {
-  // intentAction, intentCategory, intentFlags, optionalIntentArguments, dontStopAppOnReset
+  appWaitPackage, appWaitActivity, intentAction, intentCategory, intentFlags, optionalIntentArguments) {
+  // intentAction, intentCategory, intentFlags, dontStopAppOnReset
   // parameters are not supported by Espresso
   appPackage = appPackage || this.caps.appPackage;
   appWaitPackage = appWaitPackage || appPackage;
   appActivity = qualifyActivityName(appActivity, appPackage);
   appWaitActivity = qualifyActivityName(appWaitActivity || appActivity, appWaitPackage);
+  try {
+    optionalIntentArguments = (optionalIntentArguments) ? JSON.parse(optionalIntentArguments) : null;
+  } catch (e) {
+    this.log.error('Cannot parse the optionalIntentArguments JSON', e);
```

Comment:
what is the purpose in rethrowing of the exception?

Reply:
As mentioned in the issue https://github.com/appium/appium-espresso-driver/issues/802, I am passing intentoptions as JSON string and rethrowing the error to halt the execution if there are JSON parsing error and let user know that valid JSON is expected.